mitosheet/mitosheet/public/v3/sheet_functions/bool_functions.py

```diff
@@ -123,6 +124,66 @@ def IF(condition: pd.Series, true_series: AnyPrimitiveOrSeriesInputType, false_s
     )
 
 
+@handle_sheet_function_errors
+def IFS(*argv: Optional[IfsInputType]) -> pd.Series:
+    """
+    {
+        "function": "IFS",
+        "description": "Returns the value of the first condition that is true. If no conditions are true, returns None.",
+        "search_terms": ["ifs", "if", "conditional", "and", "or"],
+        "examples": [
+            "IFS(height > 100, 'tall', height > 50, 'medium', height > 0, 'short')"
+        ],
+        "syntax": "IFS(boolean_condition_1, value_if_true, [boolean_condition_2, value_if_true, ...])",
+        "syntax_elements": [{
+                "element": "boolean_condition",
+                "description": "An expression or series that returns True or False values. Valid conditions for comparison include ==, !=, >, <, >=, <=."
+            },
+            {
+                "element": "value_if_true, ... [OPTIONAL]",
+                "description": "The value the function returns if condition is True, followed by alternating boolean conditions and values."
+            }
+        ]
+    }
+    """
+    if len(argv) % 2 != 0:
+        raise MitoError(
+            'invalid_args_error',
+            'IFS',
+            'IFS requires an even number of arguments.',
+            error_modal=False
+        )
+    base_index = next(iter(s.index for s in argv if isinstance(s, pd.Series)), None)
+    # If index is None, we're dealing with all constants
+    if base_index is None:
+        for index, arg in enumerate(argv):
+            if index % 2 == 0:
+                if arg == True:
+                    return argv[index+1]
+        return None
+
+    else:
+        # Otherwise, we have at least one series -- so we can go through and turn all of the constants into series.
+        argv_series = tuple([get_series_from_primitive_or_series(arg, base_index) for arg in argv])
+        results = pd.Series(index=base_index, dtype='object')
```

Comment:
What if you made the dtype the same dtype as the first results series? I think this will fix your issues with dtypes...